aequilibrae/paths/linear_approximation.py

```diff
@@ -397,6 +397,10 @@ def derivative_of_objective(stepsize):
                     self.stepsize = 0.0
                     # need to reset conjugate / bi-conjugate direction search
                     self.do_fw_step = True
+
+                    # By doing it recursively, we avoid doing the same AoN again
```

Comment:
I took me a second to understand this, but you are right, no need to re-do the AON, although the logic is a bit more complicated now.

Reply:
Indeed